src/rgw/driver/rados/topic.cc

```diff
@@ -352,11 +352,12 @@ class MetadataHandler : public RGWMetadataHandler {
     if (r < 0) {
       return r;
     }
-    if (!info.dest.push_endpoint.empty() && info.dest.persistent) {
-      r = rgw::notify::add_persistent_topic(info.name, y);
+    if (!info.dest.push_endpoint.empty() && info.dest.persistent &&
+        !info.dest.persistent_queue.empty()) {
+      r = rgw::notify::add_persistent_topic(info.dest.persistent_queue, y);
       if (r < 0) {
         ldpp_dout(dpp, 1) << "ERROR: failed to create queue for persistent topic "
-            << info.name << " with: " << cpp_strerror(r) << dendl;
+            << info.dest.persistent_queue << " with: " << cpp_strerror(r) << dendl;
```

Comment:
line 381 also need to change and use `persistent_queue` `(use str:cat tenant + name)`

Reply:
thanks. i changed `MetadataHandler::remove()` to load the existing topic metadata so it could handle queue cleanup the same way `RGWPubSub::remove_topic()` does